src/entities/initPool/initPoolV3.ts

```diff
@@ -45,4 +43,15 @@ export class InitPoolV3 implements InitPoolBase {
             exactAmountsIn: getAmounts(tokens, input.amountsIn),
         };
     }
+
+    private value(input: InitPoolInputV3) {
+        return input.wethIsEth
+            ? (input.amountsIn.find((a) =>
+                  isSameAddress(
+                      a.address,
+                      NATIVE_ASSETS[input.chainId].address,
```

Comment:
You'll have to check for `wrapped`, since this is the input that the user has to provide.
```suggestion
                      NATIVE_ASSETS[input.chainId].wrapped,
```